### PR TITLE
add jax cudnn sdpa mla support

### PR DESCRIPTION
**Pull request by Cjkkkk**
add support for multi-head latent attention described in https://arxiv.org/pdf/2405.04434 in jax cudnn sdpa api. Main difference between mla and regular mha is that the head dim for query and value can be different.

changes made:
* remove checks for head dim equality between query and value.
* only enable this with cudnn > 9.10 and arch > hopper.

### CODE REVIEW

File: tests/fused_attention_stablehlo_test.py [hunk 1]
```diff
@@ -788,6 +788,57 @@ def train(query, key, value, grads):
...
+      value = jax.device_put(value, qkv_sharding)
+      jitted_sdpa_inference = jax.jit(
+        partial(
+          dot_product_attention, scale=1.0, mask_type=MaskType.NO_MASK,
+          dropout_rate=0),
+        in_shardings=in_shardings,
+        out_shardings=out_shardings
+      )
+
+      jitted_sdpa_inference_ref = jax.jit(
```
Comment (Rifur13) [inline tests/fused_attention_stablehlo_test.py:897]: You don't need to jit or shard the reference function.

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 2]
```diff
@@ -371,6 +374,10 @@ def check_is_flash_attention(
           raise NotImplementedError(
             "Packed layout requires cudnn version >= 9.6 and at least hopper arch.")
 
+        if is_mla and (cudnn_version < 91000 or not check_compute_capability("9.0")):
+          raise NotImplementedError(
+            "mla requires cudnn version >= 9.10 and at least hopper arch.")
```
Comment (Rifur13) [inline jax/_src/cudnn/fused_attention_stablehlo.py:401]: Why is this hopper specific?

**Review by Rifur13 — APPROVED** [review]

File: jax/_src/cudnn/fused_attention_stablehlo.py (hunk 2 above)
Comment (Cjkkkk) [reply jax/_src/cudnn/fused_attention_stablehlo.py:401]: cudnn mla is only implemented on hopper + blackwell.

File: tests/fused_attention_stablehlo_test.py (hunk 1 above)
Comment (Cjkkkk) [reply tests/fused_attention_stablehlo_test.py:897]: why shouldn't we do that? It is done for other tests.

**Comment by Cjkkkk** [comment]
I accidentally closed the PR, it is reopened now, sorry for the confusion.

File: tests/fused_attention_stablehlo_test.py (hunk 1 above)
Comment (Rifur13) [reply tests/fused_attention_stablehlo_test.py:897]: It’s fine to leave as-is, but what do we gain from testing that the sharding works on the reference function?

**Review by Rifur13 — APPROVED** [review]

**Review by justinjfu — APPROVED** [review]
